gulpfile.js

```diff
@@ -1405,12 +1412,19 @@ function createTypeScriptDefinitions() {
     .replace(/^(\s*)(export )?const enum (\S+) {(\s*)$/gm, "$1$2enum $3 {$4")
     // Replace JSDoc generation version of defined with an improved version using TS type predicates
     .replace(
-      /defined\(value: any\): boolean/gm,
-      "defined<Type>(value: Type): value is NonNullable<Type>",
+      /\n?export function defined\(value: any\): boolean;/gm,
+      `\n${readFileSync("./packages/engine/Source/Core/defined.d.ts")
+        .toString()
+        .replace(/\n*\/\*.*?\*\/\n*/gms, "")
+        .replace("export default", "export")}`,
     )
+    // Replace JSDoc generation version of Check with one that asserts the type of variables after called
     .replace(
       /\/\*\*[\*\s\w]*?\*\/\nexport const Check: any;/m,
-      `\n${readFileSync("./packages/engine/Source/Core/Check.d.ts").toString()}`,
+      `\n${readFileSync("./packages/engine/Source/Core/Check.d.ts")
+        .toString()
+        .replace(/export default.*\n/, "")
```

Comment:
The checked in `Check.d.ts` file _should_ have a newline at the end afaict but I also have no issue with making the newline optional in the regex. I've pushed up that change if you would like to take another look.

Reply:
When you scroll to the bottom of https://github.com/CesiumGS/cesium/blob/5586cf1ab83c586fd1938ce29c24b12940a948de/packages/engine/Source/Core/Check.d.ts , does it display a newline for you?

![Cesium Defined Check](https://github.com/user-attachments/assets/9393d256-d52b-47ff-bb00-bc399f8a897c)

I thought that one could alternatively look for the `;` (which kind of is a "line separator" anyhow). 
But making the `\n` optional seems to be sufficient, so that seems to be resolved now.